### PR TITLE
Add PyQt6 to tox -e mypy

### PR DESCRIPTION
**Pull request by Carreau**
This should allows mypy to infer more types.

Discussed in https://github.com/napari/napari/pull/5897#discussion_r1295538502

### CODE REVIEW

**Comment by Carreau** [comment]
@napari-bot update constraints

**Comment by Carreau** [comment]
@napari-bot update constraints

File: resources/requirements_mypy.txt [hunk 1]
```diff
@@ -2,23 +2,23 @@
 # This file is autogenerated by pip-compile with Python 3.11
 # by the following command:
 #
-#    pip-compile --config=pyproject.toml --output-file=resources/requirements_mypy.txt resources/requirements_mypy.in
```
Comment (Carreau) [inline resources/requirements_mypy.txt]: Apparently the latest pip-compile look into `pyproject.toml` automatically.

**Comment by Czaki** [comment]
@Carreau there is a bug in update dependencies workflow  #6171

**Comment by Czaki** [comment]
@napari-bot update constraints

**Comment by Czaki** [comment]
@napari-bot update constraints

**Comment by Czaki** [comment]
@napari-bot update constraints

**Comment by Czaki** [comment]
eh. I have not updated the script that was waiting on the labeled event.

**Comment by Czaki** [comment]
It looks like download from github is timing out (opengl dll for windows).

**Comment by Carreau** [comment]
Ok, cancelling all workflows and will restart them later.

**Comment by Czaki** [comment]
@napari-bot update constraints

**Comment by Czaki** [comment]
ok. So single typing error to fix and could go.

**Review by Czaki — APPROVED** [review]

**Comment by Carreau** [comment]
Conflicts resolved.

**Comment by Carreau** [comment]
Thanks !

**Pull request merged**